src/components/pomoclock/pomoclock.jsx

```diff
@@ -1,3 +1,148 @@
+import { useLocation } from "react-router-dom";
+import { CircularProgressbar } from "react-circular-progressbar";
+import "react-circular-progressbar/dist/styles.css";
+import { useEffect, useRef, useState } from "react";
+import { FaPlay, FaPause, FaStop } from "react-icons/fa";
+
 export function Pomoclock() {
-  return <>Pomoclock</>;
+  const location = useLocation();
+  const { task } = location.state;
+  const { title, description, time } = task;
+  const [seconds, setSeconds] = useState(0);
+  const [infoMessage, setInfoMessage] = useState("");
+  const [setTimeoutId, setSetTimeoutId] = useState(null);
+  const clockInitalState = {
```

Comment:
If these states are inter-dependent , then useReducer can be used to manage state.

Reply:
No, states aren't interdependent.